cupy/_core/core.pyx

```diff
@@ -60,6 +60,20 @@ from cupy.exceptions import ComplexWarning
 NUMPY_1x = numpy.__version__ < '2'
```

Comment:
Admittedly not related to the change set here, but this will speed up `true` checks

```suggestion
cdef bint NUMPY_1x = numpy.__version__ < '2'
```